fix: raise UnableToWriteHosts for privileged users

lucasfcnunes opened the pull request:
even when sudo/root editing (file = 644) it was raising this exception. simple change in edit mode w -> r+ can solve this for privileged users.

Fixes #21

lucasfcnunes commented:
maybe adding a "mode=" parameter in .write would be better... idk

jonhadfield commented:
Thanks for the PR. I appreciate you taking the time to look at this issue.
Would you be able separate this into two PRs (one for the file write issue, and the second for comments)? It just keeps it cleaner being able to comment and work on.

I've had a look at the write issue and the +r generates a write exception on Linux, and MacOS. I also tried on Windows 10 (64-bit) and use of +r also results in an error. I _suspect_ the [issue](https://github.com/jonhadfield/python-hosts/issues/34) relates to this [SO post](https://stackoverflow.com/a/13215998) and is related to 32-bit Windows only. Please could you confirm the OS (inc. architecture) and version of python you are testing this on?

lucasfcnunes commented:
*r+ opens for reading and writing (can't truncate the file)

I'm currently using this project in Linux Ubuntu (20.04, cpython 3.8) and a Alpine Docker (python:3-alpine, cpython 3.9) container as root. it was throwing the unable to write exception and this fixed for my use case.

probably the safe route is to create a new parameter "mode" in .write... i'll do that!

about the inline comments PR, i thought i hadn't pulled that: it was a mistake, sorry :(

lucasfcnunes committed:
fix: raise UnableToWriteHosts: use mode="r+"

even when sudo/root editing (file = 644) it was raising this exception
r+ can't truncate the file.
 you've been warned
